pyomeca/rototrans.py

```diff
@@ -1,499 +1,273 @@
-        angles=FrameDependentNpArray(),
-        angle_sequence="",
-        translations=FrameDependentNpArray(),
-        *args,
-        **kwargs
-    ):
+        data: Optional[Union[np.array, np.ndarray, xr.DataArray]] = None,
+        time: Optional[Union[np.array, list, pd.Series]] = None,
+        **kwargs,
+    ) -> xr.DataArray:
         """
+        Rototrans DataArray with `row`, `col` and `time` dimensions used for rototranslation matrix.
+         ![rototrans](/images/objects/rototrans.svg)
 
-        Parameters
-        ----------
-        rt : FrameDependentNpArray (4x4xF)
-            Rototranslation matrix sorted in 4x4xF, default is the matrix that don't rotate nor translate the system, is
-            ineffective if angles is provided
-        angles : FrameDependentNpArray
-            Euler angles of the rototranslation, angles parameter is ineffective if angles_sequence if not defined, but
-            will override rt
-        angle_sequence : str
-            Euler sequence of angles; valid values are all permutation of 3 axes (e.g. "xyz", "yzx", ...)
-        translations : FrameDependentNpArray
-            First 3 rows of 4th row, translation is ineffective if angles is not provided
-        """
+        Arguments:
+            data: Array to be passed to xarray.DataArray
+            time: Time vector in seconds associated with the `data` parameter
+            kwargs: Keyword argument(s) to be passed to xarray.DataArray
 
-        # Determine if we construct RotoTrans from rt or angles/translations
-        if angle_sequence:
-            rt = cls.rt_from_euler_angles(
-                angles=angles, angle_sequence=angle_sequence, translations=translations
-            )
+        Returns:
+            Rototrans `xarray.DataArray` with the specified data and coordinates
 
-        else:
-            s = rt.shape
-            if s[0] != 4 or s[1] != 4:
-                raise IndexError("RotoTrans must by a 4x4xF matrix")
-            # Make sure last line reads [0, 0, 0, 1]
-            if len(s) == 2:
-                rt[3, :] = np.array([0, 0, 0, 1])
-            else:
-                rt[3, 0:3, :] = 0
-                rt[3, 3, :] = 1
-
-        # Finally, we must return the newly created object:
-        return super(RotoTrans, cls).__new__(cls, array=rt, *args, **kwargs)
-
-    def __array_finalize__(self, obj):
-        super().__array_finalize__(obj)
-        # Allow slicing
-        if obj is None or not isinstance(obj, RotoTrans):
-            return
-
-    # --- Linear algebra methods
-
-    def get_euler_angles(self, angle_sequence):
-        """
+        !!! example
+            To instantiate a `Rototrans` 4 by 4 and 100 frames filled with some random data:
+
+            ```python
+            import numpy as np
+            from pyomeca import Rototrans
+
+            n_row = 4
+            n_col = 4
+            n_frames = 100
+            data = np.random.random(size=(n_row, n_col, n_frames))
```

Comment:
This seems wrong to me, as RT must be:
[[R3x3] [T3x1]
[0  0 0]  [   1   ]
with R3x3 being the rotation matrix which must be unitary (with all rows and lines unitary). I think creating it using from_angles would makes much more sense since would are sure to have a valid random rotation matrix

Reply:
Yes indeed, but in this example I only wanted to use `Rototrans(data)` rather than methods as this is the docstring for `__new__`.
Do you have any idea on how to make a simple example?

This feels weird:

```python
from pyomeca import Angles, Rototrans

angles = Angles.from_random_data(size=(1, 1, 100))
rt = Rototrans.from_euler_angles(angles, angle_sequence="x")

Rototrans(rt.data)
```

Reply:
This is ok, just not at the right place. If you call Rototrans.from_random_data(..) it should call Angles by itselves (so Rototrans.from_random_data() is guaranteed to produce random but valid data)

Reply:
Done in https://github.com/pyomeca/pyomeca/pull/88/commits/b22c3222efdc1138b50a58f677fd354e52190f66.